### DISCUSSION TITLE
C indentation per-project or per-directory

### DISCUSSION THREAD

**Discussion by ldelossa** [helix-editor/helix#9731] [Q&A] [answered]
Hello,

I'm starting to use helix which I like quite a bit. 

I wrote C code at work with a tab width of 8. 

My other C projects use a normal tab width of 4.

Does helix have any way to adjust this config based on either the workspace folder or parent directory of the file (keep in mind, files can be nested recursively in sub directories). 

In VSCode I tend to use 'profiles' for this. An ideal solution would not involve adding any configuration files to the workspace folder/repository itself.

**Comment by kirawi** [comment]
There is the local project config under `.helix`, but there is no way to load a language config otherwise.

**Reply by ldelossa** [reply]
Would you accept a PR which allows "per project" config to be read outside of the config dir itself? 

Can brainstorm on what that can look like id you are.

**Reply by kirawi** [reply, marked as answer]
See https://github.com/helix-editor/helix/issues/7102
